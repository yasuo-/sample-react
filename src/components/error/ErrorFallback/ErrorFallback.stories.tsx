import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import { ErrorFallback } from './ErrorFallback';

export default {
  title: 'Components/Error/ErrorFallback',
  component: ErrorFallback,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof ErrorFallback>;

const Template: ComponentStory<typeof ErrorFallback> = (args) => <ErrorFallback {...args} />;

export const Default = Template.bind({});

Default.args = {
  error: {
    name: 'Error',
    message: 'An error has occurred.',
  },
};
