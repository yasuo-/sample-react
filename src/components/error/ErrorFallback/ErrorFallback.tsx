import React from 'react';

import { FallbackProps } from 'react-error-boundary';

/**
 * ErrorFallback
 * Fallback component for ErrorBoundary.
 * @param {Error} error Error object
 * @param resetErrorBoundary Reset error boundary
 * @constructor
 */
export const ErrorFallback: React.FC<FallbackProps> = ({ error, resetErrorBoundary }) => {
  return (
    <div>
      <h2>エラーが発生しました。</h2>
      <pre>{error.message}</pre>

      <button onClick={resetErrorBoundary}>リロード</button>
    </div>
  );
};
ErrorFallback.displayName = 'ErrorFallback';
