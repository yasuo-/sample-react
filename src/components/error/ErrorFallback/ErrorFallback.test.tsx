import React from 'react';

import { render, screen } from '@testing-library/react';

import { ErrorFallback } from './ErrorFallback';

const error = {
  name: 'Error',
  message: 'An error has occurred.',
};
const resetErrorBoundary = () => jest.fn();

describe('ErrorFallback Ui Components', () => {
  it('displayName is ErrorFallback', () => {
    expect(ErrorFallback.displayName).toBe('ErrorFallback');
  });

  it('should snapshot', () => {
    const { asFragment } = render(
      <ErrorFallback error={error} resetErrorBoundary={resetErrorBoundary} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders error message text', async () => {
    render(<ErrorFallback error={error} resetErrorBoundary={resetErrorBoundary} />);

    expect(screen.getByText(error.message)).toBeInTheDocument();
  });
});
