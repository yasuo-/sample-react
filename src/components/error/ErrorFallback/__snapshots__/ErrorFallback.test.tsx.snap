// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorFallback Ui Components should snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      エラーが発生しました。
    </h2>
    <pre>
      An error has occurred.
    </pre>
  </div>
</DocumentFragment>
`;
